### PR TITLE
[fix] Enable CI for pull_requests

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

Sets up two separate GitHub Actions - 
1. on pull-request
    * checks for cpplint
    * checks compilation on Linux, MacOs and Windows
2. on the author's fork 
    * performs clang-format and commit if needed
    * update DIRECTORY.md and commit if needed
    * perform same checks as 1 on the fork

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by kvedala** [comment]
@ayaankhan98 this one is important to add please review.
Thanks

**Comment by kvedala** [comment]
@cclauss let know if you have any suggestions

File: .github/workflows/awesome_workflow.yml [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
 #     branches: [ master ]
 
 jobs:
-  code_format:
+  MainSequence:
     name: Code Formatter
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v1 # v2 is broken for git diff
+      - uses: actions/setup-python@v1
```
Comment (cclauss) [inline .github/workflows/awesome_workflow.yml:15]: v2

**Comment by kvedala** [comment]
done. please merge.

**Pull request merged**